tests/test_az.py

```diff
@@ -1,96 +1,80 @@
 """Tests for the Azure AzMultiPartUpload class."""
 
 import base64
-import unittest
 from unittest.mock import MagicMock, patch
 
-# Conditional import for Azure support
-try:
-    from odc.geo.cog._az import AzMultiPartUpload
+import pytest
 
-    HAVE_AZURE = True
-except ImportError:
-    AzMultiPartUpload = None
-    HAVE_AZURE = False
+pytest.importorskip("odc.geo.cog._az")
+from odc.geo.cog._az import AzMultiPartUpload  # noqa: E402
```

Comment:
This won't work as expected as `odc.geo.cog._az`CAN be imported now even when `azure` is missing, it just won't function after the import. I guess it should be `pytest.importorskip("azure.storage.blob")`

Reply:
thanks :), changed